DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/DllConflictManagment/DllConflictManager.cs

```diff
@@ -0,0 +1,245 @@
+
+    StringBuilder sb = new();
+    if (assemblyName != null && _assemblyConflicts.TryGetValue(assemblyName, out var assemblyConflictInfo))
+    {
+      sb.AppendLine($"Speckle encountered a dependency mismatch error.");
+      sb.AppendLine();
+      sb.AppendLine($"Dependency Name: {assemblyName}");
+      sb.AppendLine($"Expected Version: {assemblyConflictInfo.SpeckleDependencyAssemblyName.Version}");
+      sb.AppendLine($"Actual Version: {assemblyConflictInfo.ConflictingAssembly.GetName().Version}");
+      sb.AppendLine();
+      sb.AppendLine($"Conflicting dll folder: {assemblyConflictInfo.GetConflictingExternalAppName()}");
+
+      TaskDialog.Show("Conflict Report 🔥", sb.ToString());
+      return true;
+    }
+
+    return false;
+  }
+
+  private static T GetPrivateFieldValue<T>(object obj, string fieldName)
+  {
+    if (obj == null)
+    {
+      throw new ArgumentNullException("obj");
+    }
+
+    FieldInfo fi =
+      obj.GetType().GetField(fieldName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+      ?? throw new ArgumentOutOfRangeException(
+        "propName",
+        string.Format("Property {0} was not found in Type {1}", fieldName, obj.GetType().FullName)
+      );
+
+    return (T)fi.GetValue(obj);
+  }
+
+  public void WarnUserOfPossibleConflicts()
+  {
+    if (_assemblyConflicts.Count == 0)
+    {
+      return;
+    }
+
+    var options = _optionsLoader.LoadOptions();
+    List<AssemblyConflictInfo> conflictInfoNotIgnored = _assemblyConflicts.Values
+      .Where(info => !options.DllsToIgnore.Contains(info.SpeckleDependencyAssemblyName.Name))
+      .ToList();
+
+    if (conflictInfoNotIgnored.Count == 0)
+    {
+      return;
+    }
+
+    StringBuilder sb = new();
+    foreach (var assemblyConflictInfo in conflictInfoNotIgnored)
+    {
+      sb.AppendLine($"Dependency Name:  {assemblyConflictInfo.SpeckleDependencyAssemblyName.Name}");
+      sb.AppendLine($"Expected Version: {assemblyConflictInfo.SpeckleDependencyAssemblyName.Version}");
+      sb.AppendLine($"Actual Version:   {assemblyConflictInfo.ConflictingAssembly.GetName().Version}");
+      sb.AppendLine($"Conflicting version folder: {assemblyConflictInfo.GetConflictingExternalAppName()}");
+      sb.AppendLine();
+    }
+
+    using var dialog = new TaskDialog("Conflict Report 🔥");
+    dialog.MainInstruction =
+      "Speckle encountered an error loading the following dependencies. This is likely due to a different addin that uses a different version of the same dependency.";
+
+    dialog.ExtraCheckBoxText = "Do not warn about this conflict again";
+    dialog.MainContent = sb.ToString();
+    dialog.CommonButtons = TaskDialogCommonButtons.Ok;
+
+    _ = dialog.Show();
+
+    if (dialog.WasExtraCheckBoxChecked())
+    {
+      foreach (var conflictInfo in conflictInfoNotIgnored)
+      {
+        options.DllsToIgnore.Add(conflictInfo.SpeckleDependencyAssemblyName.Name);
```

Comment:
I mean this is a bit different as we're ignoring the DLL. I feel there's some versioning info missing somewhere

Reply:
I also don't follow how we're clearing this (or where atm) if the plugin fails to load ot the host app crashes (if we can detect that)

Reply:
we're not clearing the list anywhere. If the plugin initializes successfully, then we show the warning and skip the ignored conflicts. If the plugin throws an exception on initialization, we skip checking for ignored conflicts and just show the dialog